packages/safe-apps-web3-react/src/useSafeAppConnection.ts

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { useWeb3React } from '@web3-react/core';
+import { SafeAppConnector } from './connector';
+
+function useSafeAppConnection(connector: SafeAppConnector): boolean {
+  const { activate, active } = useWeb3React(); // specifically using useWeb3ReactCore because of what this hook does
+  const [tried, setTried] = React.useState(false);
+
+  React.useEffect(() => {
+    connector.isSafeApp().then((loadedInSafe) => {
+      if (loadedInSafe) {
+        activate(connector, undefined, true).catch(() => {
```

Comment:
not really required, but maybe makes it easier to understand: Should we comment tat `activate` flips the `active` flag and therefore it is not required to call `setTried(true)` on success?